### PR TITLE
feat: [PIE-4738]: service manifest overrides

### PR DESCRIPTION
**Pull request by bhavya-sinha**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 117 Issues 
* Code issue 117 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10651)

**Comment by kapilkumawat86** [comment]
trigger featurenamecheck

**Comment by kapilkumawat86** [comment]
trigger featurenamecheck

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  Bitbucket: 'Bitbucket',
+  InheritFromManifest: 'InheritFromManifest',
+  Harness: 'Harness',
+  CustomRemote: 'CustomRemote'
+}
+export const AllowedManifestOverrideTypes = [
+  OverrideManifests.Values,
+  OverrideManifests.OpenshiftParam,
+  OverrideManifests.KustomizePatches
+]
```
Comment (udayvunnam) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:39]: Aren't these available from backend code generation as enums ?

Comment (bhavya-sinha) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:39]: No

**Review by udayvunnam — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**